logspec/states/linux_kernel.py

```diff
@@ -19,9 +19,14 @@ def _detect_kernel_start(text):
     doesn't.
     """
     first_line_end = text.index('\n')
-    return re.match(fr'{LINUX_TIMESTAMP} .*',
+    result = re.match(fr'{LINUX_TIMESTAMP} .*',
                     text[:first_line_end])
 
+    if not result:
+        return re.search(fr'{LINUX_TIMESTAMP} Linux version .*',
+                         text)
+    return result
+
```

Comment:
```suggestion
    return re.match(fr'{LINUX_TIMESTAMP} .*',
                    text[:first_line_end]) or re.search(fr'{LINUX_TIMESTAMP} Linux version .*', text)
```

Lazy evaluation could help keeping it concise.

I'm not sure how to format such a long `return` statement. I'd be happy with anything `black` reformats it to.